### PR TITLE
[SPARK-42656][CONNECT][Followup] Spark Connect Shell

### PR DESCRIPTION
**Pull request by zhenlineo**
### What changes were proposed in this pull request?
Add spark connect shell to start the spark shell with spark connect enabled.
Added "-Pconnect" to build the spark connect in the distributions.
Simplified the dev shell scripts with "-Pconnect" command.

### Why are the changes needed?
Allow users to play with spark connect easily.

### Does this PR introduce _any_ user-facing change?
Yes. Added a new shell script and "-Pconnect" build option.

### How was this patch tested?
Manually tested.

### CODE REVIEW

**Comment by zhenlineo** [comment]
If this PR accepted then no need to merge https://github.com/apache/spark/pull/40303 as this PR override the changes needed there.

**Review by hvanhovell — APPROVED** [review]
LGTM - good stuff

**Comment by hvanhovell** [comment]
Merging.

File: repl/src/main/scala-2.12/org/apache/spark/repl/Main.scala [hunk 1]
```diff
@@ -121,6 +121,11 @@ object Main extends Logging {
       sparkContext = sparkSession.sparkContext
       sparkSession
     } catch {
+      case e: ClassNotFoundException if isShellSession && e.getMessage.contains(
```
Comment (LuciferYang) [inline repl/src/main/scala-2.12/org/apache/spark/repl/Main.scala:124]: https://github.com/apache/spark/pull/40318 I do a similar change for Scala 2.13

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @zhenlineo and @hvanhovell . We need to register new profile as `build_profile_flags`.
Here is the PR.
- https://github.com/apache/spark/pull/40424

**Pull request closed without merging**